pkg/config/persist.go

```diff
@@ -0,0 +1,155 @@
+package config
+
+type (
+	// Persist is how persist Klotho constructs are represented in the klotho configuration
+	Persist struct {
+		// Type represents the service used for the persist construct
+		Type string `json:"type" yaml:"type" toml:"type"`
+		// KindParams represents the set of configuration to customize the kind of persist construct represented
+		InfraParams InfraParams `json:"infra_params,omitempty" yaml:"infra_params,omitempty" toml:"infra_params,omitempty"`
+	}
+)
+
+// GetPersistKv returns the `Persist` config for the persist_kv resource specified by `id`
+// merged with the defaults.
+func (a Application) GetPersistKv(id string) Persist {
+	cfg := Persist{}
+	if ecfg, ok := a.PersistKv[id]; ok {
+		defaultParams, ok := a.Defaults.PersistKv.InfraParamsByType[ecfg.Type]
```

Comment:
There's a bunch of code duplication here that feels like we could refactor away.

Reply:
yeah in the old config we had a merge method for each kind to handle some of this, which we could do. It realistically will save about 6 lines per kind though. Fine doing a bit of a refactor, but would like to move fast and keep pushing forward since im sure well have to tweak some of the config stuff

Reply:
also because they reference different portions of the object, hard to make a reusable function